### PR TITLE
Update handling of array sizes in SplineR2R and SplineC2C for rotation cases

### PR DESCRIPTION
**Pull request by markdewing**
In the case of orbital rotation, the orbital set size from the H5 file may not match the number of SPOs used, which may not match the number of occupied orbitals.

In BsplineSet.h, the slices of the matrices need to have a size consistent with the size of the matrix.

In SplineR2R.cpp, use the size of psi rather than myV.  In the call to evaluate3d it doesn't matter because all the inputs are sized the same as myV, but in the following call to assign_*, the size discrepancy can cause invalid accesses.

Add a Python script for evaluating SPOs given the HDF file of coefficients.  It uses autograd to evaluate the spatial derivatives.

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Bugfix

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
desktop

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- Yes. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

File: src/QMCWaveFunctions/BsplineFactory/SplineC2C.cpp [hunk 1]
```diff
@@ -86,7 +86,7 @@ void SplineC2C<ST>::evaluateValue(const ParticleSet& P, const int iat, ValueVect
 #pragma omp parallel
   {
     int first, last;
-    FairDivideAligned(myV.size(), getAlignment<ST>(), omp_get_num_threads(), omp_get_thread_num(), first, last);
+    FairDivideAligned(2 * psi.size(), getAlignment<ST>(), omp_get_num_threads(), omp_get_thread_num(), first, last);
```
Comment (PDoakORNL) [inline src/QMCWaveFunctions/BsplineFactory/SplineC2C.cpp:89]: I'm not completely sure this magic number is from the same cause as all the other bare 2's in this file, but who knows. The line of documentation in the header actually manages to use the word "double" in an ambiguous way until you read the .c

**Review by PDoakORNL — CHANGES_REQUESTED** [review]
Haven't look at this file in a while. I feel kinda nauseous.  Considering the centrality of spline evaluation in the code its sad that it is basically C from the nineties with almost no documentation.

File: src/QMCWaveFunctions/BsplineFactory/SplineC2C.cpp (hunk 1 above)
Comment (prckent) [reply src/QMCWaveFunctions/BsplineFactory/SplineC2C.cpp:89]: Can we replace these 2s with a const named something about complex? I also noted the bare 2s when reading the file.

**Review by PDoakORNL — DISMISSED** [review]

**Comment by PDoakORNL** [comment]
test this please

**Review by prckent — DISMISSED** [review]

**Comment by markdewing** [comment]
Test this please

**Comment by markdewing** [comment]
Test this please

**Review by ye-luo — APPROVED** [review]

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**